src/Common/Core/Impl/OS/Win32EnvironmentBlock.cs

```diff
@@ -0,0 +1,102 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Text;
+using static Microsoft.Common.Core.NativeMethods;
+
+namespace Microsoft.Common.Core.OS {
+    public class Win32EnvironmentBlock : Dictionary<string, string> {
+        public static Win32EnvironmentBlock Create(IntPtr token, bool inherit = false) {
+            IntPtr env = IntPtr.Zero;
+            Win32EnvironmentBlock eb = new Win32EnvironmentBlock();
+            try {
+                if (CreateEnvironmentBlock(out env, token, inherit)) {
+                    IntPtr ptr = env;
+                    while (true) {
+                        string envVar = Marshal.PtrToStringUni(ptr);
+                        byte[] data = Encoding.Unicode.GetBytes(envVar);
+                        if (data.Length <= 2) {
+                            // detected double null
+                            break;
+                        } else {
+                            int idx = envVar.IndexOf('=');
```

Comment:
`Split` has an overload where you can limit the number of components. If you set it to 2, it'll split on the first one only.